sycl/CMakeLists.txt

```diff
@@ -220,20 +220,28 @@ if (MSVC)
   list(APPEND SYCL_RT_LIBS sycld)
 endif()
 
+function(add_libcxx_compile_options LIB_NAME)
+  target_compile_options(${LIB_NAME} PRIVATE -nostdinc++)
```

Comment:
Shouldn't we use libc++ for all the targets including tools and not just runtime libraries?
If so, we probably should configure `CMAKE_CXX_FLAGS` or some other variables to set these options for all the targets.

Reply:
Hi, @bader 
Previously, we wanted to investigate llvm libc++ for SYCL kernel programming and found a lot of linker error encountered when we switched to llvm libc++. After compiler team's investigation, we found building SYCL runtime with llvm libc++ can solve those linker errors. This may be why only SYCL runtime is built with llvm libc++. I remembered enabling building SYCL runtime with libc++ did cost some effort and if we want to enable building more components using llvm libc++, much more effort may be needed. @alexbatashev , please correct me if anything wrong.
If we want to use libc++ for all targets including tools per strong request, I suggest to do it in a separate PR.
Thanks very much.

Reply:
> If we want to use libc++ for all targets including tools per strong request, I suggest to do it in a separate PR.

You mean abandon this patch and replace with new implementation, right?

Reply:
Hi, @bader 
"Building the whole sycl toolchain with llvm libcxx" seems to be a 'bigger' task and needs more effort. This PR only aims to fix compiling failure issue for building SYCL runtime with llvm libcxx. If we have to enable building whole sycl toolchain with llvm libcxx per strong request, suggest to open a new PR for that.
Thanks very much.